pkg/aws/ec2/ec2_test.go

```diff
@@ -171,15 +87,29 @@ func TestCollector_Collect(t *testing.T) {
 	})
 
 	t.Run("Collect should return an error if ListOnDemandPrices returns an error", func(t *testing.T) {
```

Comment:
Nope, we _are_ testing `ListOnDemandPrices`. These tests aren't super obvious. I had to add the spot prices block here because I changed the order in the function. So if I don't mock it, spots prices fails, and code never makes it into OnDemand prices. The error we're testing is the one on line 107. `ListOnDemandPrices` is supposed to get that error and bubble it up.